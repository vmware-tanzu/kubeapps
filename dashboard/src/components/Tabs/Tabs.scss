// Copyright 2020-2022 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

.tabs {
  max-width: 100%;
  overflow-x: auto;
  overflow-y: hidden;

  .tab-button {
    &:focus {
      outline: none;
    }
  }
}
