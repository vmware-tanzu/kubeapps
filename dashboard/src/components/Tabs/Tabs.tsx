// Copyright 2020-2022 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import { useState } from "react";
import "./Tabs.scss";

interface ITabsProps {
  id: string;
  columns: Array<string | JSX.Element | JSX.Element[]>;
  data: Array<string | JSX.Element | JSX.Element[]>;
}

export default function Tabs({ id, columns, data }: ITabsProps) {
  const [selected, setSelected] = useState(0);
  const handleClick = (tab: number) => {
    return () => setSelected(tab);
  };
  return (
    <div className="tabs">
      <ul className="nav" role="tablist" id={id}>
        {columns.map((column, index) => {
          return (
            <li role="presentation" className="nav-item" key={`${column}-${index}`}>
              <button
                id={`${id}-tab${index}`}
                className={`btn btn-link nav-link tab-button ${selected === index ? "active" : ""}`}
                aria-controls={`${id}-panel${index}`}
                type="button"
                onClick={handleClick(index)}
              >
                {column}
              </button>
            </li>
          );
        })}
      </ul>
      {data.map((children, index) => {
        return (
          <section
            key={`${id}-panel${index}`}
            id={`${id}-panel${index}`}
            role="tabpanel"
            aria-labelledby={`${id}-tab${index}`}
            aria-hidden={selected !== index ? "true" : "false"}
          >
            {children}
          </section>
        );
      })}
    </div>
  );
}
