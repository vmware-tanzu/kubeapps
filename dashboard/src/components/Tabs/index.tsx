// Copyright 2020-2023 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import Tabs from "./Tabs";

export default Tabs;
