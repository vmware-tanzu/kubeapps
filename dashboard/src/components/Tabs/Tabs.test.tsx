// Copyright 2020-2022 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import { mount } from "enzyme";
import Tabs from "./Tabs";

it("renders several tabs", () => {
  const wrapper = mount(<Tabs id="tabs" columns={["foo", "bar"]} data={["FOO", "BAR"]} />);
  expect(wrapper.find("button")).toHaveLength(2);
  expect(wrapper.find("section")).toHaveLength(2);
  expect(wrapper).toMatchSnapshot();
});

it("changes content between tabs", () => {
  const wrapper = mount(<Tabs id="tabs" columns={["foo", "bar"]} data={["FOO", "BAR"]} />);
  expect(wrapper.find(".active").text()).toEqual("foo");
  expect(
    wrapper
      .find("section")
      .findWhere(s => s.prop("aria-hidden") === "false")
      .text(),
  ).toEqual("FOO");

  const otherTab = wrapper.find("#tabs-tab1");
  otherTab.simulate("click");
  wrapper.update();

  const newSelectedTab = wrapper.find("#tabs-tab1");
  expect(newSelectedTab).toHaveClassName("active");
  expect(newSelectedTab.text()).toEqual("bar");
  expect(
    wrapper
      .find("section")
      .findWhere(s => s.prop("aria-hidden") === "false")
      .text(),
  ).toEqual("BAR");
});
