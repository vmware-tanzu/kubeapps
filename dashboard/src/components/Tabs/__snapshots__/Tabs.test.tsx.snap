// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders several tabs 1`] = `
<Tabs
  columns={
    Array [
      "foo",
      "bar",
    ]
  }
  data={
    Array [
      "FOO",
      "BAR",
    ]
  }
  id="tabs"
>
  <div
    className="tabs"
  >
    <ul
      className="nav"
      id="tabs"
      role="tablist"
    >
      <li
        className="nav-item"
        key="foo-0"
        role="presentation"
      >
        <button
          aria-controls="tabs-panel0"
          className="btn btn-link nav-link tab-button active"
          id="tabs-tab0"
          onClick={[Function]}
          type="button"
        >
          foo
        </button>
      </li>
      <li
        className="nav-item"
        key="bar-1"
        role="presentation"
      >
        <button
          aria-controls="tabs-panel1"
          className="btn btn-link nav-link tab-button "
          id="tabs-tab1"
          onClick={[Function]}
          type="button"
        >
          bar
        </button>
      </li>
    </ul>
    <section
      aria-hidden="false"
      aria-labelledby="tabs-tab0"
      id="tabs-panel0"
      key="tabs-panel0"
      role="tabpanel"
    >
      FOO
    </section>
    <section
      aria-hidden="true"
      aria-labelledby="tabs-tab1"
      id="tabs-panel1"
      key="tabs-panel1"
      role="tabpanel"
    >
      BAR
    </section>
  </div>
</Tabs>
`;
